test/editor/operations-test.js

```diff
@@ -44,6 +46,74 @@ describe('editor/operations: formatting operations', () => {
+                const renderer = createRenderer();
+                const pc = createPartCreator();
+                const model = new EditorModel([
+                    pc.plain("hello ` world!"),
+                ], pc, renderer);
+
+                const range = model.startRange(model.positionForOffset(0, false),
+                    model.positionForOffset(13, false));  // hello ` world
+
+                expect(range.parts[0].text.trim().includes("`")).toBeTruthy();
+                expect(longestBacktickSequence(range.parts[0].text.trim())).toBe(1);
+                expect(model.serializeParts()).toEqual([{ "text": "hello ` world!", "type": "plain" }]);
+                formatRangeAsCode(range);
+                expect(model.serializeParts()).toEqual([{ "text": "``hello ` world``!", "type": "plain" }]);
+            });
+
+            it('escapes longer backticks in between text', () => {
+                const renderer = createRenderer();
+                const pc = createPartCreator();
+                const model = new EditorModel([
+                    pc.plain("hello```world"),
+                ], pc, renderer);
+
+                const range = model.startRange(model.positionForOffset(0, false),
+                    model.getPositionAtEnd());  // hello```world
+
+                expect(range.parts[0].text.includes("`")).toBeTruthy();
+                expect(longestBacktickSequence(range.parts[0].text)).toBe(3);
+                expect(model.serializeParts()).toEqual([{ "text": "hello```world", "type": "plain" }]);
+                formatRangeAsCode(range);
+                expect(model.serializeParts()).toEqual([{ "text": "````hello```world````", "type": "plain" }]);
+            });
+
+            it('escapes non-consecutive with varying length backticks in between text', () => {
+                const renderer = createRenderer();
+                const pc = createPartCreator();
+                const model = new EditorModel([
+                    pc.plain("hell```o`w`o``rld"),
+                ], pc, renderer);
+
+                const range = model.startRange(model.positionForOffset(0, false),
+                    model.getPositionAtEnd());  // hell```o`w`o``rld
+                expect(range.parts[0].text.includes("`")).toBeTruthy();
+                expect(longestBacktickSequence(range.parts[0].text)).toBe(3);
+                expect(model.serializeParts()).toEqual([{ "text": "hell```o`w`o``rld", "type": "plain" }]);
+                formatRangeAsCode(range);
+                expect(model.serializeParts()).toEqual([{ "text": "````hell```o`w`o``rld````", "type": "plain" }]);
+            });
+
+            it('escapes backticks in between text and untoggles already formated code text', () => {
+                const renderer = createRenderer();
+                const pc = createPartCreator();
+                const model = new EditorModel([
+                    pc.plain("`hello`world`"),
```

Comment:
This isn't valid so not great to test, can you update to
```
``hello`world``
```

Reply:
I think i'll just remove it entirely, because i have already written your suggested update. probably you missed it.

Reply:
This was the only test you had for the toggling behaviour though, we need a test which asserts that undoing the code fence also works as expected for larger (>1) fences

Reply:
so basically you're saying i should write a test for something like this: from ```hello``world``` to hello``world right?

Reply:
Yup! To make sure it doesn't regress in the future